### ISSUE TITLE
error ajenti-upgrade

### ISSUE THREAD

**Issue by legolass73** [ajenti/ajenti#1143]
I recently installed ajenti on my server (centos) and when I ask update my plugin it gives me an error message:
Upgrading Ajenti Failed:[Errno 2] No such file or directory: "/usr/local/bin/ajenti-upgrade

**Comment by ebrahimiaval** [comment]
i fixed this erro by click on upgrade button of once records

**Comment by mvkalsbeek** [comment]
I'm experiencing the same problem. There's hardly any information about 'ajenti-upgrade' to be found online.

I'm also using a fresh Ubuntu 18.04 installation.

**Comment by kiarn** [comment]
Can you please try this ( Python 2 ) : 

```
pip uninstall ajenti-panel
pip install ajenti-panel
```
The file ajenti-upgrade should be now at the right place.

**Comment by kieppie** [comment]
nope - stood up new VM/container, installed Ajenti via 1-liner, executed above uninstall+install: same results.
for `ls /usr/local/bin/` yields:
ajenti-client-ssl-gen  ajenti-panel  ajenti-ssl-gen  easy_install  easy_install-2.7  pip  pip2  pip2.7  upstart-create  upstart-reload  wheel

**Comment by kieppie** [comment]
interim solution, as root:
`
cd /usr/local/bin && wget https://raw.githubusercontent.com/ajenti/ajenti/master/ajenti-panel/ajenti-upgrade && chmod +x /usr/local/bin/ajenti-upgrade && systemctl restart ajenti.service
`

**Comment by mvkalsbeek** [comment]
> Can you please try this ( Python 2 ) :
> 
> ```
> pip uninstall ajenti-panel
> pip install ajenti-panel
> ```
> The file ajenti-upgrade should be now at the right place.

Tried that method, no success. Manually trying to install `ajenti-upgrade` through `pip` also doesn't work.   Apparently no version can be found:

```
pip install ajenti-upgrade

Collecting ajenti-upgrade
  Could not find a version that satisfies the requirement ajenti-upgrade (from versions: )
No matching distribution found for ajenti-upgrade
```

> interim solution, as root:
> `cd /usr/local/bin && wget https://raw.githubusercontent.com/ajenti/ajenti/master/ajenti-panel/ajenti-upgrade && chmod +x /usr/local/bin/ajenti-upgrade && systemctl restart ajenti.service`

This solution worked, thanks!

**Comment by nekromoff** [comment]
Confirmed with Debian 8.0
The solution below works:

> interim solution, as root:
> `cd /usr/local/bin && wget https://raw.githubusercontent.com/ajenti/ajenti/master/ajenti-panel/ajenti-upgrade && chmod +x /usr/local/bin/ajenti-upgrade && systemctl restart ajenti.service`

**Comment by kiarn** [comment]
> nope - stood up new VM/container, installed Ajenti via 1-liner, executed above uninstall+install: same results.
> for `ls /usr/local/bin/` yields:
> ajenti-client-ssl-gen ajenti-panel ajenti-ssl-gen easy_install easy_install-2.7 pip pip2 pip2.7 upstart-create upstart-reload wheel

Right, sorry, I forgot that a new version was required to solve this problem with pip.
@Eugeny threw a new version this week, and this should also work with pip.

**Comment by JohnnyHom** [comment]
Upgrade Failed returned non-zero non-zero exit status 1

cd /usr/local/bin && wget https://raw.githubusercontent.com/ajenti/ajenti/master/ajenti-panel/ajenti-upgrade && chmod +x /usr/local/bin/ajenti-upgrade && systemctl restart ajenti.service

chmod: changing permissions of '/usr/local/bin/ajenti-upgrade': Operation not permitted

**Comment by anonhostpi** [comment]
Same problem Ubuntu 23.10 (venv install)

**Comment by kiarn** [comment]
Thanks for the report, the venv path is not correctly set when launching a subprocess, I have fixed this: 280c2235a8d4a0e13296ee8a9ec77e12f4429cc8
